src/main/java/com/redhat/devtools/lsp4ij/server/definition/launching/UserDefinedLanguageServerDefinition.java

```diff
@@ -1,227 +1,230 @@
-            try {
-                initializationOptions = JsonParser.parseReader(new StringReader(initializationOptionsContent));
-            } catch (Exception e) {
-                LOGGER.error("Error while parsing JSON Initialization Options for the language server '" + getId() + "'", e);
-            }
-        }
-        return initializationOptions;
-    }
-
-    public ClientConfigurationSettings getLanguageServerClientConfiguration() {
-        if ((clientConfiguration == null) && (clientConfigurationContent != null) && !clientConfigurationContent.isBlank()) {
-            try {
-                clientConfiguration = JSONUtils.getLsp4jGson().fromJson(clientConfigurationContent, ClientConfigurationSettings.class);
-            } catch (Exception e) {
-                LOGGER.error("Error while parsing JSON client configuration for the language server '" + getId() + "'", e);
-            }
-        }
-        return clientConfiguration;
-    }
-
-    @Override
-    public @NotNull String getDisplayName() {
-        return name;
-    }
-}
+/*******************************************************************************
+ * Copyright (c) 2024 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package com.redhat.devtools.lsp4ij.server.definition.launching;
+
+import com.google.gson.JsonParser;
+import com.google.gson.annotations.SerializedName;
+import com.intellij.execution.util.ProgramParametersUtil;
+import com.intellij.openapi.project.Project;
+import com.redhat.devtools.lsp4ij.JSONUtils;
+import com.redhat.devtools.lsp4ij.client.LanguageClientImpl;
+import com.redhat.devtools.lsp4ij.client.features.LSPClientFeatures;
+import com.redhat.devtools.lsp4ij.server.StreamConnectionProvider;
+import com.redhat.devtools.lsp4ij.server.definition.ClientConfigurableLanguageServerDefinition;
+import com.redhat.devtools.lsp4ij.server.definition.LanguageServerDefinition;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.StringReader;
+import java.util.Map;
+
+/**
+ * {@link com.redhat.devtools.lsp4ij.server.definition.LanguageServerDefinition} implementation to start a
+ * language server with a process command defined by the user.
+ */
+public class UserDefinedLanguageServerDefinition extends LanguageServerDefinition implements ClientConfigurableLanguageServerDefinition {
```

Comment:
I'm not sure why this is showing up as so different here, but the only real change is to make it client-configurable.